File: 5lesson/HW/repository/UserRepository.java

import java.util.ArrayList;
import java.util.List;

import 5lesson.HW.model.User;

public class UserRepository {
	private List<User> usersList = new ArrayList<>();

	public void saveUser(User user) {
		usersList.add(user);
	}

	public void deleteUser(User user) {
		usersList.remove(user);
	}

	public void updateUser(int index, User user) {
		usersList.set(index, user);
	}

	public List<User> getAllUsers() {
		return usersList;
	}

	public void addFunds(int index, float funds) {
		User current = getAllUsers().get(index);
		current.setBalance(current.getBalance() + funds);
		updateUser(index, current);
	}

	public void removeFunds(int index, float funds) {
		User current = getAllUsers().get(index);
		if ((current.getBalance() - funds) < 0) {
			System.out.println("Баланс не может быть отрицательным!");
		} else {
			current.setBalance(current.getBalance() - funds);
			updateUser(index, current);
		}
	}
}

File: Seminar7/DZ/py/UserRepository.py
from User import User


class UserRepository:
    __usersList = []

    def __init__(self):
        self.__usersList = []

    def saveUser(self, user):
        self.__usersList.append(user)

    def deleteUser(self, user):
        self.__usersList.remove(user)

    def updateUser(self, index, user):
        self.__usersList.pop(index)
        self.__usersList.insert(index, user)

    def getAllUsers(self):
        return self.__usersList

    def addFunds(self, index, funds):
        current = self.getAllUsers()[index]
        current.setBalance(current.getBalance() + funds)
        self.updateUser(index, current)

    def removeFunds(self, index, funds):
        current = self.getAllUsers()[index]
        if (current.getBalance() - funds) < 0:
            print("Баланс не может быть отрицательным!")
        else:
            current.setBalance(current.getBalance() - funds)
            self.updateUser(index, current)
